### PR TITLE
Add STM32 I2S domain clock

### PR DESCRIPTION
**Pull request by gautierg-st**
This PR updates the STM32 I2S drivers to configure the I2S domain clock from dts instead of Kconfig.
This also allows to initialize the driver using device tree instances.
It is similar to previous PRs that also introduced domain clock.

### CODE REVIEW

**Review by erwango — COMMENTED** [review]
One question, otherwise looks good.

**Review by erwango — DISMISSED** [review]

File: drivers/i2s/i2s_ll_stm32.c [hunk 1]
```diff
@@ -108,73 +108,47 @@ static int i2s_stm32_enable_clock(const struct device *dev)
...
 		return -EIO;
 	}
 
+	if (cfg->pclk_len > 1) {
+		/* Enable I2S clock source */
+		ret = clock_control_configure(clk,
+					      (clock_control_subsys_t *) &cfg->pclken[1],
+					      NULL);
+		if (ret < 0) {
+			return -EIO;
```
Comment (FRASTM) [inline drivers/i2s/i2s_ll_stm32.c:123]: Maybe with ` LOG_ERR()  ` before returning ?

**Review by erwango — DISMISSED** [review]

File: drivers/i2s/Kconfig.stm32 [hunk 2]
```diff
@@ -22,44 +22,4 @@ config I2S_STM32_TX_BLOCK_COUNT
...
-	int "Division factor for PLLI2S VCO input clock"
-	depends on I2S_STM32_USE_PLLI2S_ENABLE
-	default 8
-	range 2 63
-	help
-	  Division factor for the audio PLL (PLLI2S) VCO input clock.
-	  PLLM factor should be selected to ensure that the VCO
-	  input frequency ranges from 1 to 2 MHz. It is recommended
-	  to select a frequency of 2 MHz to limit PLL jitter.
-	  Allowed values: 2-63
```
Comment (FRASTM) [inline drivers/i2s/Kconfig.stm32:41]: with Kconfig, there was a  valid range defined for I2S_STM32_PLLI2S_PLLM (and other PLL factors)
is the control still done elsewhere ?

Comment (gautierg-st) [reply drivers/i2s/Kconfig.stm32:41]: PLLM is either the multiplier of the main PLL, or a dedicated one for the PLLI2S.
In both cases, when configuring the PLLI2S register, we reconstruct a macro LL_RCC_PLLM_DIV_* (from the dts value) that we pass to the function. If this define does not exist (i.e. the value is invalid), we get an error.

What we don't have however is a check on the PLL N value. It is for example limited to 50-432 on F412, or 192-432 on F401 (and that means that the previous default value of 56 was invalid for F401 with no further check)

**Review by FRASTM — DISMISSED** [review]

**Comment by gautierg-st** [comment]
Rebased to resolve conflicts

**Review by FRASTM — APPROVED** [review]

File: drivers/i2s/Kconfig.stm32 (hunk 2 above)
Comment (erwango) [reply drivers/i2s/Kconfig.stm32:41]: By moving clocks from Kconfig to device tree we lost the possibility to define ranges. This is valid for all STM32 clocks and this should not be a blocker for this specific PR
In theory, it should be possible to define ranges in device tree (either using bindings, either with properties coupled with a dedicated DT macro issuing a clear error message), this should be treated as a general point.

**Review by erwango — APPROVED** [review]

**Pull request merged**